runtime/backend/interface.cpp

```diff
@@ -55,5 +55,9 @@ Error register_backend(const Backend& backend) {
   return Error::Ok;
 }
 
+Span<const Backend> get_registered_backends() {
```

Comment:
If we ever allow for runtime-registered (or deregistered) backends, this API will be very hard to support: the function must return a single, contiguous buffer containing all backends, and the caller could maintain a pointer to that buffer for the lifetime of the process.

It would be safer and more maintainable to have an API like
```
size_t get_num_registered_backends();
Result<const char*> get_backend_name(size_t index);
```

Also I think we should hide/remove the current `Backend` struct, which is really an internal detail of how the table is stored. It's used to register backends, but that actually complicates things for callers since they need to create that struct. It'd be easier for callers if the API were `register_backend(const char* name, BackendInterface* backend);`